File: Gray Code.cpp
class Solution {
public:
    vector<int> grayCode(int n) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        vector<int> res;
        if (n == 0) {
            res.push_back(0);
            return res;
        }
        n = 1 << n;
        for (int i = 0; i < n; i++) {
            res.push_back(i^(i>>1));
        }
        return res;
    }
};

File: Gray Code.cs
public class Solution {
    public IList<int> GrayCode(int n) {
        if (n == 0) {
            return new List<int> {0};
        }
        
        var result = GrayCode(n - 1);
        var added = 1 << (n - 1);
        for (var i = result.Count - 1; i >= 0; i--) {
            result.Add(result[i] + added);
        }
        
        return result;
    }
}
